go/consensus/tendermint/seed/seed.go

```diff
@@ -92,49 +294,56 @@ func New(dataDir string, identity *identity.Identity, genesisProvider genesis.Pr
 	// to get the PEX reactor to operate in seed mode.
 
 	srv := &seedService{
-		quitCh: make(chan struct{}),
+		quitCh:   make(chan struct{}),
+		identity: identity,
 	}
 
 	seedDataDir := filepath.Join(dataDir, "tendermint-seed")
-	if err = common.InitDataDir(seedDataDir); err != nil {
+	if err = tmcommon.InitDataDir(seedDataDir); err != nil {
 		return nil, fmt.Errorf("tendermint/seed: failed to initialize data dir: %w", err)
 	}
 
 	p2pCfg := config.DefaultP2PConfig()
-	p2pCfg.AllowDuplicateIP = true
 	p2pCfg.SeedMode = true
-	p2pCfg.AddrBookStrict = !viper.GetBool(common.CfgDebugP2PAddrBookLenient)
-	// MaxNumInboundPeers/MaxNumOutboundPeers
+	p2pCfg.Seeds = strings.ToLower(strings.Join(viper.GetStringSlice(tmcommon.CfgP2PSeed), ","))
+	p2pCfg.ExternalAddress = viper.GetString(tmcommon.CfgCoreExternalAddress)
+	p2pCfg.MaxNumInboundPeers = viper.GetInt(tmcommon.CfgP2PMaxNumInboundPeers)
+	p2pCfg.MaxNumOutboundPeers = viper.GetInt(tmcommon.CfgP2PMaxNumOutboundPeers)
+	p2pCfg.SendRate = viper.GetInt64(tmcommon.CfgP2PSendRate)
+	p2pCfg.RecvRate = viper.GetInt64(tmcommon.CfgP2PRecvRate)
+	p2pCfg.AddrBookStrict = !(viper.GetBool(tmcommon.CfgDebugP2PAddrBookLenient) && cmflags.DebugDontBlameOasis())
+	p2pCfg.AllowDuplicateIP = viper.GetBool(tmcommon.CfgDebugP2PAllowDuplicateIP) && cmflags.DebugDontBlameOasis()
 
 	nodeKey := &p2p.NodeKey{PrivKey: crypto.SignerToTendermint(identity.P2PSigner)}
 
 	doc, err := genesisProvider.GetGenesisDocument()
 	if err != nil {
 		return nil, fmt.Errorf("tendermint/seed: failed to get genesis document: %w", err)
 	}
+	srv.doc = doc
 
 	nodeInfo := p2p.DefaultNodeInfo{
 		ProtocolVersion: p2p.NewProtocolVersion(
-			version.P2PProtocol,
-			version.BlockProtocol,
-			0,
+			tmversion.P2PProtocol,
+			tmversion.BlockProtocol,
+			version.ConsensusProtocol.ToU64(),
```

Comment:
Nice catch! @tjanez don't forget about this one in #3360 (depends on which PR gets merged first).